spec/requests/atmosphere/api/virtual_machine_flavors_spec.rb

```diff
@@ -143,13 +143,13 @@
         end
 
         it 'returns one flavor with memory greater or equal to preference memory specified in at' do
-          tmpl = create(:virtual_machine_template, appliance_type: at, tenant: t2, state: 'active')
+          tmpl = create(:virtual_machine_template, appliance_type: at, tenants: [t2], state: 'active')
           get api("/virtual_machine_flavors?appliance_type_id=#{at.id}", user)
           flavors = fls_response
           expect(flavors.size).to eq 1
           fl = flavors.first
           expect(fl['memory']).to be >= required_mem
-          expect(fl['compute_site_id']).to eq tmpl.tenant_id
+          expect(tmpl.tenants.collect{|t| t.id}).to include fl['compute_site_id']
```

Comment:
Using `pluck` you can execute only one SQL query without need of traversing through collection:

```ruby
tmpl.tenants.pluck(:id)
```